### DISCUSSION TITLE
Error in the post-build script

### DISCUSSION THREAD

**Discussion by malakm2002** [codemagic-ci-cd/codemagic-docs#1879] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

i am building a flutter application as an implementation for a community service project and while trying to release my Flutter app to App Center for beta testing, I got the error.

and this is my post-build script:
```
echo 'Installing App Center CLI tools'
    npm install -g appcenter-cli
    echo "Publishing $apkPath to App Center"
    appcenter distribute release \
        --group "Beta testers" \
        --file $apkPath\
        --release-notes 'App submission via Codemagic' \
        --app malakm2002/JuvJustAcc-App \
        --token $APP_CENTER_TOKEN \
        --quiet

```
where malakm2002 is my github username

### Expected results

i expected that the app will be on app center after the android build was succesfull

### Actual results

i got an error while running the post build script: 
```
Installing App Center CLI tools
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated formidable@1.2.6: Please upgrade to latest, formidable@v2 or formidable@v3! Check these notes: https://bit.ly/2ZEqIau
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated vm2@3.9.19: The library contains critical security issues and should not be used for production! The maintenance of the project has been discontinued. Consider migrating your code to isolated-vm.
npm WARN deprecated superagent@5.1.0: Please upgrade to v7.0.2+ of superagent.  We have fixed numerous issues with streams, form-data, attach(), filesystem errors not bubbling up (ENOENT on attach()), and all tests are now passing.  See the releases tab for more information at <https://github.com/visionmedia/superagent/releases>.

added 510 packages in 41s

83 packages are looking for funding
  run `npm fund` for details
Publishing build/app/outputs/apk/release/app-release.apk to App Center
Error: Failed to create release upload for build/app/outputs/apk/release/app-release.apk. Backend response: API token does not belong to the requested app.


Build failed :|
Step 6 script `Post-build script` exited with status code 3
```

### Build id (optional)

64b3d2ec0b7039ffb6445f00

**Comment by malakm2002** [comment]
kindly note that i followed the following tutorial step by step:
[https://blog.codemagic.io/how-to-publish-flutter-apps-to-appcenter/](url)

**Comment by kulakarla** [comment]
Hello @malakm2002 

I tried to reproduce - I followed the guide step-by-step aswell and managed to publish it to App Center easily. 
The error seems rather self-explanatory: API token does not belong to the requested app.
Make sure that you are the App Token you uploaded to Codemagic Environment Variables does indeed belong to the same app you are trying to publish. Go to your JuvJustAcc-App settings -> App API tokens -> New API Token. Grant it full access. Add it to environment variables. Maybe you accidentally deleted a letter from the end of the token or something along the lines.

Let me know if you were to run into any more problems.

**Comment by kulakarla** [comment]
I am tenatively closing this thread due to inactivity.
